sstables/checksummed_data_source.cc

```diff
@@ -17,6 +17,7 @@
 #include "exceptions.hh"
 #include "checksum_utils.hh"
 #include "data_source_types.hh"
+#include "checksummed_data_source.hh"
```

Comment:
How did we get away without having this include so far?

Reply:
What's the problem here? Definition without declaration? I don't see `-Wmissing-prototypes` in flags.

Reply:
No problem per se, I'm just wondering how the code compiled without this header was included. Seems like a missing include. Maybe I'm missing something. Anyway, I was just thinking out loud.

Reply:
Prior to this PR, the header provided the declarations of the two factory methods, so it was only used by the consumers of the API.

With this PR, it also contains the type definition for the error handler, which is necessary in the function definitions as well.